tests/system/marathon_common_tests.py

```diff
@@ -1241,3 +1243,174 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+       fault domain as the master.
+    """
+
+    # Discover which is the "local" cluster domain
+    local_domain = get_cluster_local_domain()
+
+    # Get how many agents do we have in the local domain (zone and region) and try to start
+    # that many applications. Mesos should round-robbin the instances in the region
+    region_agents = count_agents_in_faultdomains(regions=local_domain.region)
+
+    # Create an app with default configuration
+    app_def = apps.faultdomain_app(instances=region_agents, suffix="-defaults")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region (it's ok if they belong to different zones)
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(local_domain.region,))
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_region_only():
+    """Tests if the applications that only have a `region` specified are launched on all the different
+       zones available in that region on the cluster
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region ({})".format(
+        region,
+    )
+
+    # Create an app pinned on the given region, with at least double the number of instances than
+    # the zones in the cluster, to ensure that they are distributed among all of them
+    app_def = apps.faultdomain_app(region=region, instances=len(zones) * 2, suffix="-region-only")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(region,), zones=zones)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_region_and_zone():
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region ({})".format(
+        region,
+    )
+
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # (We are starting more instances than the number of zones in the region in order to increase
+    #  the chance of a faulty instance to appear on a different zone)
```

Comment:
Super nit. Why the `(` and `)`?

Reply:
It's a guilty comment. `( )` denotes something we _shouldn't_ be doing, but we are doing nonetheless :trollface:

Reply:
Who know that?